runtime/program_params_validation_test.go

```diff
@@ -1391,4 +1391,95 @@ func TestRuntimeTransactionParameterTypeValidation(t *testing.T) {
 		expectRuntimeError(t, err, &ArgumentNotImportableError{})
 	})
 
+	newEnumType := func() cadence.Enum {
+		return cadence.NewEnum([]cadence.Value{
+			cadence.NewInt(0),
+		}).WithType(cadence.NewEnumType(
+			common.AddressLocation{
+				Address: common.MustBytesToAddress([]byte{0x1}),
+				Name:    "C",
+			},
+			"C.Alpha",
+			cadence.IntType,
+			[]cadence.Field{
+				{
+					Identifier: sema.EnumRawValueFieldName,
+					Type:       cadence.IntType,
+				},
+			},
+			nil,
+		))
+	}
+
+	t.Run("Enum Optional Type", func(t *testing.T) {
+		t.Parallel()
+
+		contracts := map[common.AddressLocation][]byte{
+			{
+				Address: common.MustBytesToAddress([]byte{0x1}),
+				Name:    "C",
+			}: []byte(`
+		   access(all) contract C {
+			   access(all) 
+			   enum Alpha: Int {
+				  access(all)
+				  case A
+
+				  access(all)
+				  case B
+			  }
```

Comment:
Just a nit-pick: seems there are mix indentations of whitespaces and tabs. Maybe use only one kind (e.g: whitespaces only) for indentations.

Reply:
Switched to tabs for consistency.